File: examples/jasonette/extensions/ios/console-action/JasonConsoleAction.m
//
//  JasonConsoleAction.m
//  Jasonette
//
//  Created by e on 1/9/17.
//  Copyright © 2017 Jasonette. All rights reserved.
//

#import "JasonConsoleAction.h"

@implementation JasonConsoleAction
- (void)debug{
    NSDictionary *variables = [[Jason client] variables];
    NSMutableArray *vars = [[NSMutableArray alloc] init];
    
    if(self.options){
        if(self.options[@"eval"]){
            [vars addObject:@{@"name": @"evaluated", @"value": self.options[@"eval"]}];
        }
    }
    
    for(NSString *key in variables){
        [vars addObject:@{@"name": key, @"value": variables[key]}];
    }
    for(NSString *key in [JasonMemory client]._register){
        [vars addObject:@{@"name": key, @"value": [JasonMemory client]._register[key]}];
    }
    
    
    NSDictionary *href = @{
                           @"type": @"$href",
                           @"options": @{
                               @"url": @"file://jasonconsoleaction.json",
                               @"options": @{@"variables": vars}
                           }
   };
   [[Jason client] call: href];
}
@end

File: examples/jasonette/extensions/android/console-action/JasonConsoleAction.java
package com.jasonette.seed.Action;

import android.content.Context;
import android.util.Log;

import com.jasonette.seed.Core.JasonViewActivity;

import org.json.JSONArray;
import org.json.JSONObject;

import java.util.Iterator;

public class JasonConsoleAction {
    public void debug(final JSONObject action, JSONObject data, Context context) {
        try {
            if(action.has("options")){
                final JSONObject options = action.getJSONObject("options");
                JSONArray kvarrary = new JSONArray();

                if (options.has("eval")) {
                    JSONObject kv = new JSONObject();
                    kv.put("name", "evaluated");
                    kv.put("value", options.get("eval"));
                    kvarrary.put(kv);
                }

                JSONObject href = new JSONObject();
                JSONObject params = new JSONObject();
                JSONObject new_options = new JSONObject();

                Iterator<String> keysIterator = data.keys();
                try {
                    while (keysIterator.hasNext()) {
                        String key = (String) keysIterator.next();
                        Object val = data.get(key);
                        JSONObject kv = new JSONObject();
                        kv.put("name", key);
                        kv.put("value", val);
                        kvarrary.put(kv);
                    }
                } catch (Exception e) {

                }
                params.put("variables", kvarrary);
                new_options.put("url", "file://jasonconsoleaction.json");
                new_options.put("options", params);
                href.put("type", "$href");
                href.put("options", new_options);
                ((JasonViewActivity)context).call(href.toString(), "{}", context);
            }
        } catch (Exception e) {
            Log.d("Error", e.toString());
        }
    }
}
